weaver/main.go

```diff
@@ -87,6 +89,14 @@ func main() {
 		routerName = iface.HardwareAddr.String()
 	}
 
+	if nickName == "" {
+		osHostname, err := os.Hostname()
+		if err != nil {
+			log.Fatal(err)
+		}
+    	nickName = osHostname
+	}
+
```

Comment:
The indentation has gone rather wrong in the above. Also, might as well assign straight to nickName.